### DISCUSSION TITLE
Deployment Query + Enhancement with Docker

### DISCUSSION THREAD

**Discussion by xmagcx** [devfinwiz/Fin-Maestro#24] [Q&A]
Hi, good, I want to know which version of python is compatible, I already tried with 3.8 and 3.9 and it throws me a lot of errors with the requirements.txt libraries. Also if it is possible within the MD to indicate how to raise the platform because there is no documentation in this regard

**Comment by devfinwiz** [comment]
Hey, thanks for reaching out. Most of the functionalities should function with python 3.8 and 3.9 but modules like Crypto need 3.10+ version. I would suggest using python 3.10.11 as it works well for me. Consider upgrading your python version before trying to get going with testing out Fin-maestro. 

For talib library installation you can refer to this: https://blog.quantinsti.com/install-ta-lib-python/ as it cannot be downloaded directly with pip.

**Reply by xmagcx** [reply]
I understand, thanks for the clarification, regarding my second question, how the project itself is built, since I did not find the front part to do it.  I think a little tutorial in the MD file would be good about it, if I can get up, I can support you to be able to take the initiative through Docker containers and thus be able to make the project more scalable in case you are interested

**Reply by devfinwiz** [reply]
Yes, containerization for Fin-Maestro is one of our aims for the future. I currently have too much on my plate and hence this won't be done in the near future. 

The front end part of the code (Vue.Js) is in the branch withFE. And the other alternative (which can be executed within minutes) of the front end part of the code is in the master branch under CommonFEGradio/driver.py.

**Comment by devfinwiz** [comment]
Also, if you wish to contribute by containerizing Fin-Maestro, it would be a great help. Feel free to contribute by creating a new branch. Thanks.

**Reply by xmagcx** [reply]
perfect when I can raise it locally I'll start to create the image. Can you give me permission to create a new branch from main?

**Reply by xmagcx** [reply]
Hello dev, I already have the fork called docker, for now I managed to dockerize the back, but I did a restructuring of both the folders and parts of the code since some standard is missing. Now I will continue with the front part. I would like to propose that you review the code in this fork and listen to your comments. Leave some TODO's to review too
![image](https://github.com/devfinwiz/Fin-Maestro/assets/20447334/f533fa4d-4d40-42c9-9c0c-6ec38988c28e)

**Reply by devfinwiz** [reply]
Hey, that looks great! I am a little busy till the 22nd of June. You carry on with the development until then locally. I'll soon get you a branch where you can push your changes.

**Reply by xmagcx** [reply]
Hello, how are you? ready with the docker both back and front, I also made a change in the MD to incorporate the installation method

![image](https://github.com/devfinwiz/Fin-Maestro/assets/20447334/1f260b61-1653-4d5a-8a8a-0266776dde5f)

**Reply by devfinwiz** [reply]
Great job mate! I'll get in touch with you by the 22nd of June. Meanwhile, can you share the demo of how the execution flow looks like in the form of a video or sequence of images? You can spam it all on my email devjuneja43@gmail.com .

**Reply by devfinwiz** [reply]
Hey @xmagcx , I've created a new branch docker. You can use it as your destination branch while raising the PR instead of the master. Also, pull this branch on your fork and for any further developments use it and raise PRs. 
https://github.com/devfinwiz/Fin-Maestro/tree/docker

**Reply by xmagcx** [reply]
perfect i'll PR with my fork

**Reply by devfinwiz** [reply]
Hey, I've merged your PR. I wanted to know if you have thoroughly tested by brining the project up? Is everything functional post containerization? I was planning to merge the containerized branch to main after creating a copy of the old master into a new branch.

**Reply by devfinwiz** [reply]
I won't be able to spare much time for further development of Fin-Maestro like before.

**Reply by xmagcx** [reply]
Hi, how are you?? I did the necessary tests of each endpoint and it works quite well, but I think it would be important not to depend on reading CSV files, you could consider using some postgres to manage that information. On the other hand, I think it would be prudent to use the image from the docker hub repository, I don't know if you have an account? Mine is very basic and the backend image is quite heavy due to the amount of AI libraries, that could also be reduced by using some hugging face (I have no experience, it's a proposal)

**Reply by xmagcx** [reply]
In the meantime, I could dedicate myself to getting everything to postgres, but I don't have experience in frontend and datascience to debug the modules

**Reply by devfinwiz** [reply]
Hey, Sounds good. The csv files are used by just 1 or 2 modules at max. For all other modules, there is provision of sending a request to Yfinance everytime for data. Although, this is also inefficient without a proper caching mechanism. Will work towards it sometime in future (not in near future unfortunately).And, yes you can help me shift towards postgres for the selected modules that presently use csv. Will let you know about that as well soon.

**Reply by xmagcx** [reply]
Yes of course I can help you with the postgres thing (I can advance a few weekends) and we will also leave it as a container
